src/auto_scheduler/search_task.cc

```diff
@@ -66,6 +69,11 @@ HardwareParams HardwareParamsNode::GetDefaultHardwareParams(const Target& target
 
     device_api->GetAttr(ctx, tvm::runtime::DeviceAttrKind::kMaxRegistersPerBlock, &ret);
     int max_registers_per_block = ret;
+    if (max_registers_per_block == 0) {
+      LOG(INFO) << "Device API reported kMaxRegistersPerBlock being 0, using the value of "
+                   "kMaxSharedMemoryPerBlock as max_registers_per_block.";
```

Comment:
It seems to me that most people won't understand the meaning of this message. Maybe either turning it to a comment, or mentioning this is for the performance cost model?

Reply:
This is following mercy's suggestion https://discuss.tvm.apache.org/t/auto-schedular-performance-on-amdgpu-the-first-attempt/8582/17

Since I also don't understand this issue well, I turned it into a comment with more info.